src/categories/categories.service.ts

```diff
@@ -72,4 +73,26 @@ export class CategoriesService {
       )
     }
   }
+
+  async deleteCategory(categoryId: string, generalCateroryId: string){
+    try {
+      await this.categoryRepo.delete(categoryId)
+
+      //move all apis in this category to a general category section
+      await this.apiRepo.createQueryBuilder()
+        .update()
+        .set({categoryId: generalCateroryId})
+        .where('categoryId = :categoryId', {categoryId})
+        .execute()
+      
```

Comment:
This logic won't work. You need to update first before deleting, else you get foreign key constraint error.
You need to get an array of all apis in that category, then do something like `.where({ id: In(apiIds) })`

Please test your code.

Reply:
The code has been tested several times and it works completely. 

But it makes sense to update before deleting first.